### PR TITLE
Upgrade to Rails 7

### PR DESCRIPTION
**Pull request by brendanshean**
## WHAT
Upgrade our LMS to Rails 7

## WHY
Rails 7 offers various improvements to Rails 6

## HOW
1.  [Disable](https://guides.rubyonrails.org/active_record_multiple_databases.html#granular-database-connection-switching) legacy connection handling
1.  Replace `db:structure:load` with `db:schema:load` as it is [deprecated](https://github.com/rails/rails/commit/5488686851978c64346a0c03d2b6541cf2a94879#diff-1ed2907b3b8f148c2533558a77673ffa)
1.  Add explicit render of user_mailer view templates as they are no longer autoloaded.
1.  Fix DEPRECATION WARNING: TimeWithZone#to_s(:db) 

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  No.  Configuration change.
Have you deployed to Staging? | YES
Self-Review: Have you done an initial self-review of the code below on Github? | YES
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? |  N/A

### CODE REVIEW

File: services/QuillLMS/bin/setup [hunk 1]
```diff
@@ -1,31 +1,35 @@
...
+  system("bundle check") || system!("bundle install")
 
-  unless File.exist?("config/database.yml")
-    puts "\n== Copying sample files =="
-    system "cp config/database.yml.sample config/database.yml"
-  end
+  # puts "\n== Copying sample files =="
+  # unless File.exist?("config/database.yml")
+  #   FileUtils.cp "config/database.yml.sample", "config/database.yml"
+  # end
```
Comment (anathomical) [inline services/QuillLMS/bin/setup]: Should we just delete this?

File: services/QuillLMS/app/queries/scorebook/query.rb [hunk 2]
```diff
@@ -118,7 +118,7 @@ def self.date_conditional_string(begin_date, end_date, offset)
   end
 
   def self.to_offset_datetime(date, offset)
-    (Date.parse(date).midnight - offset.seconds).to_s(:db)
+    (Date.parse(date).midnight - offset.seconds).to_fs(:db)
```
Comment (anathomical) [inline services/QuillLMS/app/queries/scorebook/query.rb:121]: I believe that we also use `to_s(:db)` in `app/queries/snapshots/period_query.rb`.  Not sure if that's the same issue as here, but figured I'd flag it.

**Review by anathomical — APPROVED** [review]
Excited that this is moving along.

File: services/QuillLMS/bin/setup (hunk 1 above)
Comment (brendanshean) [reply services/QuillLMS/bin/setup]: Good idea

File: services/QuillLMS/app/queries/scorebook/query.rb (hunk 2 above)
Comment (brendanshean) [reply services/QuillLMS/app/queries/scorebook/query.rb:121]: Good call.  I was trying to figure out why there were no deprecation warnings after this fix, then I remembered that I was not looking at the correct [configuration](https://app.circleci.com/pipelines/github/empirical-org/Empirical-Core/23417/workflows/4c498ea0-d9c9-438e-99b2-f21ad04e67ac/jobs/280247?invite=true#step-108-1681)
(over 1000 warnings!)

**Pull request merged**